storage/streams/src/main/java/io/apicurio/registry/streams/events/KafkaEventSink.java

```diff
@@ -0,0 +1,103 @@
+
+import java.time.Instant;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.UUID;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.common.header.internals.RecordHeaders;
+import org.apache.kafka.common.serialization.Serdes;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.apicurio.registry.events.EventSink;
+import io.apicurio.registry.utils.RegistryProperties;
+import io.apicurio.registry.utils.kafka.AsyncProducer;
+import io.apicurio.registry.utils.kafka.ProducerActions;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.eventbus.Message;
+
+/**
+ * @author Fabian Martinez
+ */
+@ApplicationScoped
+public class KafkaEventSink implements EventSink {
+
+    private static final Logger log = LoggerFactory.getLogger(KafkaEventSink.class);
+
+    @Inject
+    @RegistryProperties(
+            value = {"registry.streams.common", "registry.streams.events"},
+            empties = {"ssl.endpoint.identification.algorithm="}
+    )
+    Properties producerPproperties;
+
+    private ProducerActions<UUID, byte[]> producer;
+    private Integer partition;
+
+    @ConfigProperty(name = "registry.events.kafka.topic")
+    Optional<String> eventsTopic;
+
+    @ConfigProperty(name = "registry.events.kafka.topic-partition")
+    Optional<Integer> eventsTopicPartition;
+
+    @PostConstruct
+    void init() {
+        partition = eventsTopicPartition.orElse(0);
+    }
+
+    @Override
+    public String name() {
+        return "Kafka Sink";
+    }
+
+    @Override
+    public boolean isConfigured() {
+        return eventsTopic.isPresent();
+    }
+
+    @Override
+    public void handle(Message<Buffer> message) {
+        String type = message.headers().get("type");
+
+        log.info("Firing event " + type);
+
+        UUID uuid = UUID.randomUUID();
+
+        Headers headers = new RecordHeaders();
+        headers.add("ce_id", uuid.toString().getBytes());
+        headers.add("ce_specversion", "1.0".getBytes());
+        headers.add("ce_source", "apicurio-registry".getBytes());
+        headers.add("ce_type", type.getBytes());
+        headers.add("ce_time", Instant.now().toString().getBytes());
+        headers.add("content-type", "application/json".getBytes());
+
+        getProducer()
+            .apply(new ProducerRecord<UUID, byte[]>(
+                    eventsTopic.get(),
+                    partition,
+                    uuid,
+                    message.body().getBytes(),
+                    headers));
+
+    }
+
+    public synchronized ProducerActions<UUID, byte[]> getProducer() {
+        if (producer == null) {
+            producer = new AsyncProducer<UUID, byte[]>(
+                    producerPproperties,
```

Comment:
Where are the rest of Kafka properties defined? BOOTSTRAP_SERVERS_CONFIG, Auth, etc.

Reply:
they are picked from the main apicurio registry configuration. This means, events will be sent to the same kafka using the same connection config, authentication/authorization config as the registry is using for the storage

Reply:
That works for us but it may be a problem for someone using a different storage but they want to use Kafka to replicate schemas. I'm not sure if that's something you would like to approach in this stage. It works as it is for us

Reply:
great! I don't think it's necessary to add that possibility at this point. I'll prefer to wait until someone has a requirement for that.

Reply:
I disagree - I think we'll definitely want the ability to send messages to Kafka even if the storage variant is e.g. SQL.

Reply:
Also for many deployments that are using the Streams storage, the Kafka cluster used for registry storage may be different from the one handling registry events.

Reply:
@famartinrh the original idea was send all events to the EventBus from storage and every consumer which is registered and configured can receive it and send to the destination regarding implemented technology i.e. Kafka, http, cloud event, etc.

Reply:
ok, I reorganized things so kafka events are available for all registry variants